app/models/wiki.rb

```diff
@@ -73,14 +74,18 @@ def api_url
   #############
 
   def ensure_valid_project
-    # Multilingual projects must have language == nil.
+    # Most multilingual projects must have language == nil,
+    # but wikisource is the exception, it can have nil, or a language.
     # Standard projects must have a language.
-    if MULTILINGUAL_PROJECTS.include?(project)
+    #
+    # TODO: Validate the language/project combination by pinging it's API.
+    if project == "wikidata"
       self.language = nil
+    elsif project == "wikisource"
```

Comment:
Note that translating `www` to `nil` here keeps the behavior consistent with the `Wiki` spec we added earlier: 'for projects with a multilingual version in addition to language support'.

What we don't want is two Wiki records that represent the same wiki, for example, one with `www` and one with `nil` for the language. The spec should probably be tweaked to make sure that when you create one, and then `.get_or_create` the other, it returns the first one.